numpy/_core/multiarray.py

```diff
@@ -1362,11 +1362,11 @@ def shares_memory(a, b, max_work=None):
         of candidate solutions to consider). The following special
         values are recognized:
 
-        max_work=MAY_SHARE_EXACT  (default)
+        max_work=-1 (formerly MAY_SHARE_EXACT)
```

Comment:
Let's not include "formerly", it doesn't age well.  Also the `(default)` is good to keep IMO.